CMakeLists.txt

```diff
@@ -29,38 +29,6 @@ else ()
       "Compile shared libraries (if OFF, static libraries are compiled)." ON)
 endif()
 
-# Detect whether the user passed BUILD_PYTHON_BINDINGS in order to determine if
```

Comment:
Hi @mikkac, these configuration are at right place. You have to target the setup present in `src/mlpack/CMakeLists.txt` i.e.
```Cmake
if (BUILDING_PYTHON_BINDINGS)
...
endif()

if (BUILDING_JULIA_BINDINGS)
...
endif()

if (BUILDING_GO_BINDINGS)
...
endif()
```

Reply:
Right, exactly.  Sorry @mikkac if the issue was unclear---the part to refactor is not this bit, but the stuff in `src/mlpack/CMakeLists.txt` that's kind of like `post_markdown_setup()`.  In fact you could see `post_markdown_setup()` as "factored out", and so the idea is to take the stuff that's in `src/mlpack/CMakeLists.txt` at the end and related to each individual binding type, and then just put that in functions with exactly the name that you've got here.  Let me know if I can clarify that. :+1:  And thanks for taking the time to work on this! :rocket:

Reply:
Hi, sure- I could misunderstand the issue's description, no problem :) I will take those tips into consideration and try to work something out in free time- sorry, currently I'm very busy with other projects. Stay tuned!

Reply:
Sounds good---no worries.  Good luck with the other projects! :+1: